### PR TITLE
Feature update architecture

### PR DESCRIPTION
**Pull request by Lagyamfi**
I have tried to create basic architecture to be followed using the code from Prince as the base. 
I have added some tests as well and done some clean up in the code.

### CODE REVIEW

File: khaya/tts_api.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import json
+from requests.models import Response
+
+from khaya.abstract_api import BASE_API
+
+
+class TTSAPI(BASE_API):
```
Comment (joelbudu) [inline khaya/tts_api.py]: Can we make these CapWords case? like TtsApi and BaseAPi

File: khaya/asr_api.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from requests.models import Response
+
+from khaya.abstract_api import BASE_API
+
+
+class ASRAPI(BASE_API):
```
Comment (joelbudu) [inline khaya/asr_api.py]: Can we make this CapWords case?

File: khaya/tts_api.py (hunk 1 above)
Comment (Lagyamfi) [reply khaya/tts_api.py]: updates made. Ref commit `7afb86d34cf0e8ebc223824a8e383acbb3e72397`

File: khaya/asr_api.py (hunk 2 above)
Comment (Lagyamfi) [reply khaya/asr_api.py]: updates made. Ref commit `7afb86d34cf0e8ebc223824a8e383acbb3e72397`

**Comment by Lagyamfi** [comment]
@joelbudu thanks for the review. Updates made as requested. If you are happy, can you merge it. I want to start to work on the documentation.
Thanks

**Review by joelbudu — APPROVED** [review]

**Pull request merged**